packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx

```diff
@@ -296,34 +297,41 @@ export let DataSpreadsheet = React.forwardRef(
 
     useEffect(() => {
       const currentHeaders: Array<any> = [];
-      const pastHeaders: Array<any> = [];
       if (Object.keys(currentColumns).length > 0) {
         Object.keys(currentColumns).forEach((itemIndex) => {
           if (currentColumns[itemIndex].Header) {
             currentHeaders.push(currentColumns[itemIndex].Header);
           }
         });
       }
-      if (Object.keys(columns).length > 0) {
-        Object.keys(columns).forEach((itemIndex) => {
-          if (columns[itemIndex].Header) {
-            pastHeaders.push(columns[itemIndex].Header);
-          }
-        });
+
+      if (previousState.selectedHeaderReorderActive) {
+        setPastColumns(currentHeaders);
       }
 
       if (
         !previousState.selectedHeaderReorderActive &&
         !headerCellHoldActive &&
         currentHeaders.length > 0 &&
-        pastHeaders.length > 0 &&
-        JSON.stringify(currentHeaders) !== JSON.stringify(pastHeaders)
+        pastColumns.length > 0 &&
+        JSON.stringify(currentHeaders) !== JSON.stringify(pastColumns)
       ) {
```

Comment:
Hey @AustinGitHub ,
Thanks for the changes and those looks good. But I am seeing many duplicate lines of code in the `if` block and `else if` block. Is it possible to do some refactor?

Reply:
Sure

Reply:
I fixed it, thanks!